Sources/PackageModel/UserToolchain.swift

```diff
@@ -117,15 +117,81 @@ public final class UserToolchain: Toolchain {
         return toolPath
       }
     }
+
+    if useXcrun {
+      #if os(Windows)
+        func getHostTriple(from destination: Destination, swiftCompilerPath: AbsolutePath?) -> Triple? {
+          if let triple = destination.hostTriple {
+            return triple
+          } else if let swiftCompilerPath = swiftCompilerPath {
+            return .getHostTriple(usingSwiftCompiler: swiftCompilerPath)
+          } else {
+            return nil
+          }
+        }
+        if let hostTriple = getHostTriple(from: destination, swiftCompilerPath: swiftCompilerPath),
+           let tool = findToolInVisualStudio(name, triple: destination.targetTriple, hostTriple: hostTriple) {
+          return tool
+        }
+      #endif
+    }
     throw InvalidToolchainDiagnostic("could not find \(name)")
   }
 
+  private static func findToolInVisualStudio(_ name: String, triple: Triple?, hostTriple: Triple) -> AbsolutePath? {
+    // Returns MSVC names for the triple arch
+    func vcArchNames(triple: Triple) -> (component: String, host: String, target: String)? {
+      guard triple.isWindows() else { return nil }
+      switch triple.arch {
+      case .x86_64:
+        return ("Microsoft.VisualStudio.Component.VC.Tools.x86.x64", "HostX64", "x64")
+      case .i686:
+        return ("Microsoft.VisualStudio.Component.VC.Tools.x86.x64", "HostX86", "x86")
+      case .arm64, .arm64e, .aarch64:
+        return ("Microsoft.VisualStudio.Component.VC.Tools.ARM64", "HostARM64", "arm64")
+      case .arm, .armv5, .armv6, .armv7:
```

Comment:
This is absolutely incorrect.  Anything but `.armv7` should be a hard error.  LLVM does not support WinCE, WinNT does not support anything prior to ARMv7, and even then requires Thumb2.

Reply:
Thanks for pointing out, will fix later